src/registration/meanSquareError.ts

```diff
@@ -0,0 +1,57 @@
+import imagePairAndOffset from "./imagePairAndOffset";
+
+export function meanSquareError(args: imagePairAndOffset) {
+    let diff = commonAreaDifference(args);
+    let total = 0;
+    for (let i = 0; i < diff.length; i++) {
+        for (let j = 0; j < diff[0].length; j++) {
```

Comment:
Should be `diff[i].length`.
Once rotation is introduced, each row will be in different length.

Or better, use 1-dimensional array.